### ISSUE TITLE
📎 Implement `lint/noEmptyBlockStatements` - `eslint/no-empty` `eslint/no-empty-static-block`

### ISSUE THREAD

**Issue by Conaclos** [biomejs/biome#39]
### Description

This lint rule should integrate both [no-empty](https://eslint.org/docs/latest/rules/no-empty/) and [no-empty-static-block](https://eslint.org/docs/latest/rules/no-empty-static-block/) rules.

**Want to contribute?** Lets you know you are interested! We will assign you to the issue to prevent several people to work on the same issue. Don't worry, we can unassign you later if you are no longer interested in the issue! Read our [contributing guide](https://github.com/biomejs/biome/blob/main/CONTRIBUTING.md) and [analyzer contributing guide](https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md).

**Comment by ematipico** [comment]
What's the difference between this rule and https://github.com/biomejs/biome/issues/42? It's worth an explanation in the description

**Comment by Conaclos** [comment]
> What's the difference between this rule and #42? It's worth an explanation in the description

In contrast to #42, the rule prohibits any empty block statements.
Examples of incorrect code:

```js
if (cond) {}

while (cond) {}

function f() {}
```

However, the following example also trigger #42:

```js
{}
```

**Comment by ematipico** [comment]
So we would have **two** rules that trigger the same code? That doesn't seem right to me. 

I wonder if at this point if we should have only one rule that handles all the block statements. Or at least make sure that they don't conflict each other.

**Comment by Conaclos** [comment]
> I wonder if at this point if we should have only one rule that handles all the block statements

I think you are right. We could thus regroup these two rule under the `noUselessBlockStatements` name.

**Comment by andrewangelle** [comment]
Hello! If a fix for this issue is still needed, I'd like to give it a shot.
